### PR TITLE
[4] webservices custom field list and radio

### PR DESCRIPTION
**Pull request by alikon**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/38642 .

### Summary of Changes
redo of ex-#38655

### Testing Instructions

have a list with multiple yes and or a radio articol field
set some articles

`GET {{base_path}}/api/index.php/v1/content/articles`

### Actual result BEFORE applying this Pull Request
error 500

### Expected result AFTER applying this Pull Request
no more  error 500

### CODE REVIEW

**Comment by antonbag** [comment]
I have tested this item :white_check_mark: successfully on 3819ddf415ca7b08dfcedc3be2ee4780237e83a6<br /><br />Now It works :)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41777">issues.joomla.org/tracker/joomla-cms/41777</a>.</sub>

**Comment by MacJoom** [comment]
@alikon - Thanks for the PR. What is the test setup for radio buttons - i can see the error 500 with the multiple items in list type and the bug is fixed - but i cannot force the error with the radio buttons

**Comment by alikon** [comment]
create a radio field make it not required and not fill it in an article

without pr 

you get 
`  "serie-a": {
                    "": null
                },`

with the pr 

`"serie-a": [],`

File: plugins/fields/list/src/Extension/ListPlugin.php [hunk 1]
```diff
@@ -44,9 +44,14 @@ public function onCustomFieldsBeforePrepareField($context, $item, $field)
             return;
         }
 
-        $options = $this->getOptionsFromField($field);
+        $field->apivalue = [];
+        $options         = $this->getOptionsFromField($field);
```
Comment (Quy) [inline plugins/fields/list/src/Extension/ListPlugin.php]:
```suggestion
        $options         = $this->getOptionsFromField($field);
        $field->apivalue = [];
```

Comment (Quy) [reply plugins/fields/list/src/Extension/ListPlugin.php]: For consistency with radio.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on e012e727e0324e37f8eeaf99c90e6d7a3d26570a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/41777">issues.joomla.org/tracker/joomla-cms/41777</a>.</sub>

**Comment by MacJoom** [comment]
Thank you!

**Pull request merged**